src/classes/HealthCheckDefinitionService_TEST.cls

```diff
@@ -0,0 +1,93 @@
+* @description Test class for HealthCheckDefinitionService
+*/
+@isTest
+private class HealthCheckDefinitionService_TEST {
+    
+    /**************************************************************************************************************************
+    * @description Test method to verify that the getInstance method returns a new instance of the 
+    * HealthCheckDefinitionService class when one does not already exist.
+    ***************************************************************************************************************************/
+    @isTest
+    private static void getInstanceNew() {
+        Test.startTest();
+            HealthCheckDefinitionService healthCheckDefinitionServiceInstance = HealthCheckDefinitionService.getInstance();
+        Test.stopTest();
+
+        System.assertEquals(HealthCheckDefinitionService.instance, healthCheckDefinitionServiceInstance, 'Instance of service class returned should match static instance variable.');
+        System.assertEquals(true, healthCheckDefinitionServiceInstance != null, 'Instance of service class should not be null.');
+    }
+
+    /**************************************************************************************************************************
+    * @description Test method to verify that the getInstance method returns the existing instance of the 
+    * HealthCheckDefinitionService class when one already exists.
+    ***************************************************************************************************************************/
+    @isTest
+    private static void getInstanceExisting() {
+        Test.startTest();
+            HealthCheckDefinitionService healthCheckDefinitionServiceInstance1 = HealthCheckDefinitionService.getInstance();
+            HealthCheckDefinitionService healthCheckDefinitionServiceInstance2 = HealthCheckDefinitionService.getInstance();
+        Test.stopTest();
+
+        System.assertEquals(HealthCheckDefinitionService.instance, healthCheckDefinitionServiceInstance1, 'Instance of service class returned should match static instance variable.');
+        System.assertEquals(healthCheckDefinitionServiceInstance1, healthCheckDefinitionServiceInstance2, 'Subsequent retrievals of service class instance should return existing instance.');
+        System.assertEquals(true, healthCheckDefinitionServiceInstance1 != null, 'Instance of service class should not be null.');
+    }
+
+    /**************************************************************************************************************************
+    * @description Test method to verify that the getHealthCheckDefinitionModels method in Health Check Service class
+    * returns a list of Health Check Definition models
+    ***************************************************************************************************************************/
+    @isTest
+    private static void getHealthCheckDefinitionModelsExistingFromService() {
+        Test.startTest();
```

Comment:
Stub out the HealthCheckDefinitionMapper here to always return an expected two or more HealthCheckDefinitionModels. Then test that those expected values are returned.

I'd prefer we start using stubbing early to take advantage of the singleton pattern.